presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1996,18 +1996,76 @@ VeloxQueryPlanConverterBase::toVeloxQueryPlan(
   auto insertTableHandle =
       std::make_shared<core::InsertTableHandle>(connectorId, hiveTableHandle);
 
-  auto outputType = toRowType(
-      {node->rowCountVariable,
-       node->fragmentVariable,
-       node->tableCommitContextVariable});
+  std::vector<protocol::VariableReferenceExpression> variables = {
+      node->rowCountVariable,
+      node->fragmentVariable,
+      node->tableCommitContextVariable};
+
+  std::shared_ptr<core::AggregationNode> aggregationNode;
+  if (node->statisticsAggregation) {
+    auto outputVariables = node->statisticsAggregation->outputVariables;
+    auto aggregationMap = node->statisticsAggregation->aggregations;
+    auto groupingVariables = node->statisticsAggregation->groupingVariables;
+    VELOX_CHECK_EQ(
+        aggregationMap.size(),
+        outputVariables.size(),
+        "TableWriterNode's aggregationMap and outputVariables should be the same size");
+    VELOX_CHECK_GT(
+        outputVariables.size(),
+        0,
+        "TableWriterNode's outputVariables shouldn't be empty");
+    variables.insert(
+        variables.end(), groupingVariables.begin(), groupingVariables.end());
+    for (auto const& variable : outputVariables) {
+      variables.push_back(variable);
+    }
+
+    std::vector<std::string> aggregateNames;
+    std::vector<core::AggregationNode::Aggregate> aggregates;
+    std::vector<core::FieldAccessTypedExprPtr> aggrMasks;
+    aggregateNames.reserve(aggregationMap.size());
+    aggregates.reserve(aggregationMap.size());
+    for (const auto& entry : outputVariables) {
+      aggregateNames.emplace_back(entry.name);
+      core::AggregationNode::Aggregate aggregate;
+      aggregate.call = std::dynamic_pointer_cast<const core::CallTypedExpr>(
+          exprConverter_.toVeloxExpr(aggregationMap.at(entry).call));
+      if (aggregationMap.at(entry).mask != nullptr) {
+        aggregate.mask =
+            exprConverter_.toVeloxExpr(aggregationMap.at(entry).mask);
+      }
+      if (aggregationMap.at(entry).orderBy) {
+        for (const auto& orderBy : aggregationMap.at(entry).orderBy->orderBy) {
```

Comment:
Not sure if we can just check order by is not set for column stats aggregation? Is it used in Java side?